bpfd/src/bpf.rs

```diff
@@ -333,9 +337,88 @@ impl BpfManager {
                     Err(BpfdError::UnableToPinProgram(e))
                 })?;
 
+            Ok(id)
+        } else if let Program::Uprobe(ref program) = p {
```

Comment:
We should be able to streamline the add_single_attach_program() function and have less duplicated code, but I decided that I'd tackle that when I handle the other probe types.

Reply:
https://github.com/bpfd-dev/bpfd/issues/564